tools/xmlschema.rb

```diff
@@ -121,18 +156,14 @@ def printDocumentation(_file, _spaces, _doc)
 end
 
 #################################################
-def printIncludeRef(_file, _spaces, _inc)
-  path = File.join($path, _inc.attributes["filename"])
-  doc = REXML::Document.new File.new(path)
-  incElemName = doc.root.attributes['name']
-  _file.printf("%*s<xsd:element ref='%s'/>\n", _spaces, "", incElemName)
-end
-
-#################################################
+# Prints XSD contents of <include>
 def printInclude(_file, _spaces, _attr)
-  loc = "http://sdformat.org/schemas/"
-  loc += _attr.attributes['filename'].sub("\.sdf","\.xsd")
-  _file.printf("%*s<xsd:include schemaLocation='%s'/>\n", _spaces, "", loc)
+  loc = $path + "/" + _attr.attributes['filename']
+  doc = REXML::Document.new File.new(loc)
+
+  doc.elements.each_with_index("element") do |elem, i|
+    printXSD(_file, _spaces+2, elem)
```

Comment:
I think if we should use `printElem` instead of `printXSD` here  so that the `pose` element gets the `pose` type. Right now this is what it generates

```xml
<xsd:element name='pose'>
  <xsd:annotation>
    <xsd:documentation xml:lang='en'>
      <![CDATA[A position(x,y,z) and orientation(roll, pitch yaw) with respect to the frame named in the relative_to attribute.]]>
    </xsd:documentation>
  </xsd:annotation>
  <xsd:complexType mixed='true'>
    <xsd:attribute name='relative_to' type='xsd:string'  >
      <xsd:annotation>
        <xsd:documentation xml:lang='en'>
          <![CDATA[
Name of frame relative to which the pose is applied.
]]>
        </xsd:documentation>
      </xsd:annotation>
    </xsd:attribute>
  </xsd:complexType>
</xsd:element>
```

(1) We have `<xsd:complexType mixed='true'>` which indicate that `<pose>` can have child elements, which is not currently true
(2). The xsd type `pose` is not used at all so it is not validating based on the regex. You can try this by trying to validate <pose>1.0</pose>` in an sdf file. It passes currently.

When using `printElem`, it generates:
```xml
<xsd:element name='pose'>
  <xsd:annotation>
    <xsd:documentation xml:lang='en'>
      <![CDATA[A position(x,y,z) and orientation(roll, pitch yaw) with respect to the frame named in the relative_to attribute.]]>
    </xsd:documentation>
  </xsd:annotation>
  <xsd:complexType>
    <xsd:simpleContent>
      <xsd:extension base='pose'>
        <xsd:attribute name='relative_to' type='xsd:string'  >
          <xsd:annotation>
            <xsd:documentation xml:lang='en'>
              <![CDATA[
Name of frame relative to which the pose is applied.
]]>
            </xsd:documentation>
          </xsd:annotation>
        </xsd:attribute>
      </xsd:extension>
    </xsd:simpleContent>
  </xsd:complexType>
</xsd:element>

```

This extends the `pose` type, so validation works properly.

Reply:
Thanks! 75b33b6d6447182b5945832d8296cf883f55c908